### PR TITLE
Fix e2e test that depends on env

### PR DESCRIPTION
**Pull request by mortent**
This fixes some e2e tests that relied on specific error messages, but that was dependent on the local setup. This instead stubs out the `RunE` function on the Cobra command. This also required moving the check for required setters from `RunE` to `PreRunE`, but that is the right place to put this check.
It also updates the `go.sum` file as a result of running `make tidy`.

### CODE REVIEW

File: e2e/e2e_test.go [hunk 1]
```diff
@@ -601,3 +612,21 @@ openAPI:
 		})
 	}
 }
+
+func findSubCmd(cmd *cobra.Command, names ...string) *cobra.Command {
```
Comment (droot) [inline e2e/e2e_test.go:616]: you can probably can use https://godoc.org/github.com/spf13/cobra#Command.Find

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -559,7 +561,7 @@ openAPI:
           required: true
           isSet: true
  `,
-			errMsg: `Package uninitialized`,
+			errMsg: `completed`,
```
Comment (droot) [inline e2e/e2e_test.go:564]: reading the name field in the above two test cases suggests the test should fail with some error message ?

File: e2e/e2e_test.go (hunk 1 above)
Comment (mortent) [reply e2e/e2e_test.go]: I looked for something like this, but for some reason I didn't find it. I have updated the PR to use the built-in cobra function.

File: e2e/e2e_test.go (hunk 2 above)
Comment (mortent) [reply e2e/e2e_test.go]: Yeah, the name and the expected error is a bit misleading here. I have updated the PR to just expect no error in the last two test cases.

**Review by droot — APPROVED** [review]

**Pull request merged**